### PR TITLE
Simpler printing for Iterators.Enumerate, Iterators.Zip, and SkipMissing

### LINKED ISSUE #33695: Better printing of `zip` output
The printing of zip is quite unreadable for even simple arguments:
```julia
julia> zip((:a,:b,:c), (1,2,3))
Base.Iterators.Zip{Tuple{Tuple{Symbol,Symbol,Symbol},Tuple{Int64,Int64,Int64}}}(((:a, : :c), (1, 2, 3)))
```
Can we include better printing of `zip` outputs so that for interactive sessions, it's easier to scan?

### PR DESCRIPTION
**Pull request by sostock**
Fixes #33695.

This changes the printing of `Iterators.Enumerate` and `Iterators.Zip` so they are printed just as they are called:

Before:
```julia
julia> enumerate([1,2,3])
Base.Iterators.Enumerate{Array{Int64,1}}([1, 2, 3])

julia> zip([1,2,3], ('a', 'b', 'c'))
Base.Iterators.Zip{Tuple{Array{Int64,1},Tuple{Char,Char,Char}}}(([1, 2, 3], ('a', 'b', 'c')))
```
After:
```julia
julia> enumerate([1,2,3])
enumerate([1, 2, 3])

julia> zip([1,2,3], ('a', 'b', 'c'))
zip([1, 2, 3], ('a', 'b', 'c'))
```

### CODE REVIEW

**Comment by nalimilan** [comment]
Thanks. As the CI failure shows, you'll have to update docstrings to reflect the new output.

**Review by StefanKarpinski — APPROVED** [review]

File: base/show.jl [hunk 1]
```diff
@@ -2223,6 +2223,15 @@ function showarg(io::IO, r::ReinterpretArray{T}, toplevel) where {T}
     print(io, ')')
 end
 
+# printing iterators from Base.Iterators
+
+function show(io::IO, e::Iterators.Enumerate)
```
Comment (fredrikekre) [inline base/show.jl:2228]: Should this be three-arg show?

Comment (nalimilan) [reply base/show.jl:2228]: IIUC 2-arg `show` is required to be parseable, but both the old and the new outputs satisfy this criterion, so probably better use the most compact form. See #34734.

Comment (sostock) [reply base/show.jl:2228]: If we use three-arg `show`, then `print` will use the old printing. Is that the desired behavior?

Comment (KristofferC) [reply base/show.jl:2228]: As Milan says, since this still parses it should be fine to use as 2 arg i think.

**Comment by sostock** [comment]
I added the same printing for `skipmissing` (cf. https://github.com/JuliaLang/julia/issues/33695#issuecomment-610280399).

**Comment by sostock** [comment]
I’m not sure why the CI doesn’t work for the latest commit …

**Review by nalimilan — APPROVED** [review]

**Pull request merged**